### PR TITLE
Exception handleing when loading mod options

### LINKED ISSUE #7232: Crash whenever mods menu is opened for some time
**Platform**
Android

**Version**
4.1.12 patch 1, downloaded from Github

**Describe the bug**
Crash to Desktop when opening mods menu from main menu, sometimes it's immediate, sometimes it takes a while. Can't sort them either, using any kind of sorting option. Right side mods list also doesn't show all mods, as if it's trying to load more but failed.
There is no error message.

**To Reproduce**
1. Open mods menu
2. Wait for the mods to load, crash may happen
3. Try to sort mods, doesn't work
4. Crash when trying to load more mods

**Expected behavior**
No crash

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by alexban011**
resolves  #7232

In that issue the mod had `[ ]` and failed to load the mod options. Should we automatically remove these brackets or let the modders fix this themselves ?

### CODE REVIEW

**Comment by alexban011** [comment]
on a second thought I'm not sure anymore if this is actually gonna do anything. This function is already called inside a try catch

File: core/src/com/unciv/ui/pickerscreens/GitHub.kt [hunk 1]
```diff
@@ -319,13 +319,25 @@ object Github {
      */
     fun rewriteModOptions(repo: Repo, modFolder: FileHandle) {
         val modOptionsFile = modFolder.child("jsons/ModOptions.json")
-        val modOptions = if (modOptionsFile.exists()) json().fromJsonFile(ModOptions::class.java, modOptionsFile) else ModOptions()
+        var modOptions = ModOptions()
+
+        try {
+            modOptions = if (modOptionsFile.exists()) json().fromJsonFile(ModOptions::class.java, modOptionsFile) else ModOptions()
+        } catch (ex: Throwable) {
+            Log.error("Mod options were not read correctly.")
```
[the remark this answers is not in the document]
Comment (alexban011) [reply core/src/com/unciv/ui/pickerscreens/GitHub.kt:327]: Well I'm instantiating the ModOptions above as if it entered the else branch (aka if there is no modOptions.json) so the program should be able to continue on, right? 

I just managed to get the android version this issue is occurring in (Android 11) and of a mod which had this issue. I'm gonna do some tests now.

File: core/src/com/unciv/ui/pickerscreens/GitHub.kt [hunk 2]
```diff
@@ -319,13 +319,25 @@ object Github {
      */
     fun rewriteModOptions(repo: Repo, modFolder: FileHandle) {
         val modOptionsFile = modFolder.child("jsons/ModOptions.json")
-        val modOptions = if (modOptionsFile.exists()) json().fromJsonFile(ModOptions::class.java, modOptionsFile) else ModOptions()
+        var modOptions = ModOptions()
+
+        try {
+            modOptions = if (modOptionsFile.exists()) json().fromJsonFile(ModOptions::class.java, modOptionsFile) else ModOptions()
+        } catch (ex: Throwable) {
```
[the remark this answers is not in the document]
Comment (alexban011) [reply core/src/com/unciv/ui/pickerscreens/GitHub.kt:326]: Noted. I thought it better to catch Throwable

File: core/src/com/unciv/ui/pickerscreens/GitHub.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexban011) [reply core/src/com/unciv/ui/pickerscreens/GitHub.kt]: I can't get master to crash so I can't test if this prevents that. I replaced `Throwable` with `Exception` and some specific exceptions that I thought might occur but yeah.. I can't properly test this.. 

I'm gonna try on an older phone of mine to see if maybe it crashes there (android 9)

[the remark this answers is not in the document]
Comment (alexban011) [reply core/src/com/unciv/ui/pickerscreens/GitHub.kt]: > Have you tested it with that?

yes and worst that happens is the mod doesn't download. Tried on desktop, android 12 and 9 and I can't get it to crash (not even the master or GP, let alone the PR) 

> The correct approach

I'll try this, but what would a correct exception handling be? Cancel downloading the mod? It's really annoying that I can't reproduce and therefor test this. I'm pretty sure these mods had the `[ ]` but only started failing recently. Was there any change related to this which caused the issue? I can't tell why it even worked before. 

Also, if the mod options fail to load, should the mod be disabled/not downloaded ?

> the origin and owner and stuff are still written

they are. Technically I can remove the if branch and just try to deserialize into the modOptions var and if that fails it still has the instantiated object from above.

**Comment by alexban011** [comment]
Looks like every call of rewriteModOptions has a try-catch. Only this one was missing.

**Review by yairm210 — APPROVED** [review]

**Pull request merged**